src/prefect/concurrency/asyncio.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+from typing import Awaitable, Callable, List, Literal, Union
+from contextlib import asynccontextmanager
+
+import httpx
+import pendulum
+
+from prefect import get_client
+from .events import (
+    _emit_concurrency_acquisition_events,
+    _emit_concurrency_release_events,
+)
+from prefect.client.schemas.responses import MinimalConcurrencyLimitResponse
+from prefect.utilities.math import bounded_poisson_interval, clamped_poisson_interval
+from prefect.settings import PREFECT_CLIENT_RETRY_JITTER_FACTOR
+
+
+async def wait_for_successful_response(
+    fn: Callable[..., Awaitable[httpx.Response]],
+    *args,
+    max_retry_seconds=30,
+    retryable_status_codes=[429, 502, 503],
```

Comment:
Nitpicking: we could run into issues with mutable default args right? I don't think it has any effect in practice since `retryable_status_codes` is never altered in the function

Reply:
Good point, it should be fine now and I don't see why we would ever alter the list in the function, but better to just not have the possibility.